### PR TITLE
Bug 1554293 - logging-eventrouter event not formatted correctly in El…

### PR DESCRIPTION
**Pull request by nhosoi**
…asticsearch when using MUX

Disable process_kubernetes_events if TRANSFORM_EVENTS is false or MUX client.

Depends on https://github.com/openshift/openshift-ansible/pull/10207

### CODE REVIEW

**Comment by nhosoi** [comment]
WIP - adding a test case...

**Comment by nhosoi** [comment]
Unfortunately, the current fix is incomplete.  It does not pass my to-be-added-test cases. :(

File: test/eventrouter.sh [hunk 1]
```diff
@@ -27,8 +33,31 @@ else
     esopssvc=$( get_es_svc es-ops )
     esopssvc=${esopssvc:-$essvc}
 
+    # Make sure there's no MUX
+    oc set env ds/logging-fluentd MUX_CLIENT_MODE- 2>&1 | artifact_out
+    oc label node --all logging-infra-fluentd-
+    oc label node --all logging-infra-fluentd=true
```
Comment (richm) [inline test/eventrouter.sh]: the `oc set env` has the effect of restarting fluentd - if that is sufficient in this case, you can get rid of the `oc label node` and add something like this, to wait for fluentd to be up and running.  That is, the usual idiom is  - label node with `logging-infra-fluentd-` to shut off fluentd, then wait for the daemonset numberReady to be 0.  Then, edit the fluentd ds, configmap, whatever changes are needed to fluentd.  Then, relabel the node, and wait for a fluentd pod to be running:
```
    # undeploy fluentd
    oc label node --all logging-infra-fluentd- 2>&1 | artifact_out
    os::cmd::try_until_text "oc get daemonset logging-fluentd -o jsonpath='{ .status.numberReady }'" "0" $FLUENTD_WAIT_TIME
    # oc set env, edit ds, whatever
    oc label node --all logging-infra-fluentd=true 2>&1 | artifact_out
    os::cmd::try_until_text "oc get pods -l component=fluentd" "^logging-fluentd-.* Running "
```

**Comment by richm** [comment]
Finally, the eventrouter test is not running:
```
[INFO] Logging test suite test-eventrouter started at Tue Sep 25 06:15:35 UTC 2018
No resources found.
[WARNING] Eventrouter not deployed
[INFO] Logging test suite test-eventrouter succeeded at Tue Sep 25 06:15:35 UTC 2018
```

**Comment by richm** [comment]
/lgtm
on hold until https://github.com/openshift/openshift-ansible/pull/10207 merges

**Comment by nhosoi** [comment]
Thank you, @richm .  I pushed the new patch based on your suggestion.  (I just wanted to make sure Fluentd is restarted with the valid MUX_CLIENT_MODE...)

Regarding the issue eventrouter is not enabled, I thought "-e openshift_logging_install_eventrouter=True" was passed to ansible...  Where I can enable it?

**Comment by nhosoi** [comment]
> > Regarding the issue eventrouter is not enabled, I thought "-e openshift_logging_install_eventrouter=True" was passed to ansible... Where I can enable it?
> 
> https://github.com/openshift/aos-cd-jobs/tree/master/sjb/config/test_cases - but it is tricky - I think if you re-enable `-e openshift_logging_install_eventrouter=True` for the "base" jobs like test_cases/test_branch_openshift_ansible_logging.yml and test_cases/test_branch_openshift_ansible_logging_json_file.yml it will apply to _all_ branches. If that is ok, then you could just look for all places where `openshift_logging_install_eventrouter` is commented out and add them back.

Indeed, it's tricky...  I'd like to enable the eventrouter test in 3.9 and above.   To do so, I'm supposed to discomment from "# -e openshift_logging_install_eventrouter=True" and move it to the right place in all these files?  (I don't see _39_ or older yml files in the aos-cd-jobs/sjb/config/test_cases directory.)
```
test_branch_openshift_ansible_logging_310.yml
test_branch_openshift_ansible_logging_json_file_310.yml
test_branch_openshift_ansible_logging_json_file.yml
test_branch_openshift_ansible_logging.yml
test_pull_request_origin_aggregated_logging_journald_310.yml
test_pull_request_origin_aggregated_logging_journald.yml
```
Thanks.

**Comment by nhosoi** [comment]
@richm , eventrouter was disabled with this comment.   I wonder the bug has been already solved?
> logging - workaround ansible eventrouter recursion bug

**Comment by richm** [comment]
> @richm , eventrouter was disabled with this comment. I wonder the bug has been already solved?

I think so - I think it was an ansible version problem

**Comment by nhosoi** [comment]
> > @richm , eventrouter was disabled with this comment. I wonder the bug has been already solved?
> 
> I think so - I think it was an ansible version problem

Thank you, @richm !!  I've submit this PR.
https://github.com/openshift/aos-cd-jobs/pull/1594

**Comment by nhosoi** [comment]
bot, retest this please

**Comment by nhosoi** [comment]
Thanks, @richm .  Removed the hold label.

**Pull request merged**